camel/agents/paper_to_code_agent.py

```diff
@@ -0,0 +1,856 @@
+            return [self.remove_spans(item) for item in data]
+        return data
+
+    def _process(self):
+        r"""Process the paper file by cleaning up its JSON structure.
+        
+        Reads the paper file, removes unnecessary spans from the JSON data,
+        and writes the cleaned data back to the file.
+        """
+        with open(f'{self.file_path}') as f:
+            data = json.load(f)
+        cleaned_data = self.remove_spans(data)
+        with open(self.file_path, 'w') as f:
+            json.dump(cleaned_data, f)
+
+    def planning(self, input_message: str = ""):
+        r"""Generate a plan for reproducing the paper's methodology and experiments.
+        
+        This phase creates a detailed roadmap for implementation by analyzing the
+        paper content and outputting a structured plan, architecture design, and
+        logic design.
+        
+        Args:
+            input_message (str, optional): Additional input to incorporate into the planning
+                prompts. (default: "")
+        """
+        with open(self.file_path) as f:
+            paper_content: str = f.read()
+
+        responses = []
+        trajectories = []
+        trajectories.append({
+            'role': 'system',
+            'content': self.planning_system_message.content
+        })
+        for user_message_template in _PLANING_USER_PROMPTS:
+            context = {
+                "paper_content": paper_content or "",
+                "input_message": input_message or "",
+            }
+            template = Template(user_message_template)
+            user_message = template.render(**context)
+            trajectories.append({'role': 'user', 'content': user_message})
+
+            completion = super().step(user_message)
+            completion_json = json.loads(completion.model_dump_json())
+            responses.append(completion_json)
+            message = completion.msg
+            trajectories.append({
+                'role': 'assistant',
+                'content': message.content
+            })
+            logger.info(f'planning response: {message}')
+
+        os.makedirs(self.output_path, exist_ok=True)
+        with open(f'{self.output_path}/planning_response.json', 'w') as f:
+            json.dump(responses, f)
+        with open(f'{self.output_path}/planning_trajectories.json', 'w') as f:
+            json.dump(trajectories, f)
+
+    def _extract_config(self) -> None:
+        r"""Extract configuration information from the planning output.
+        
+        Processes the planning trajectories to extract YAML configuration,
+        architecture design, and logic design, saving them as separate artifacts.
+        """
+        with open(f'{self.output_path}/planning_trajectories.json',
+                  encoding='utf8') as f:
+            traj = json.load(f)
+
+        yaml_raw_content = ""
+        for turn_idx, turn in enumerate(traj):
+            if turn_idx == 8:
```

Comment:
```suggestion
            if turn_idx == len(traj):
```
Maybe it would be better if we don't use hardcode here?